crates/ruff_python_semantic/src/all.rs

```diff
@@ -69,10 +71,10 @@ impl Ranged for DunderAllDefinition<'_> {
 /// Extract the names bound to a given __all__ assignment.
 ///
 /// Accepts a closure that determines whether a given name (e.g., `"list"`) is a Python builtin.
-pub fn extract_all_names<F>(stmt: &Stmt, is_builtin: F) -> (Vec<DunderAllName>, DunderAllFlags)
-where
-    F: Fn(&str) -> bool,
-{
+pub fn extract_all_names<'a>(
```

Comment:
Nit: We could also consider making it a method on Semantic model (I think that's what you recommended initially)